src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs

```diff
@@ -169,5 +170,30 @@ public IActionResult GetAddressesBalances(string addresses, int minConfirmations
                 return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
             }
         }
+
+
+        /// <summary>Provides verbose balance data of the given addresses.</summary>
+        /// <param name="addresses">A comma delimited set of addresses that will be queried.</param>
+        /// <returns>A result object containing the balance for each requested address and if so, a meesage stating why the indexer is not queryable.</returns>
+        [Route(BlockStoreRouteEndPoint.GetVerboseAddressesBalances)]
+        [HttpGet]
+        public IActionResult GetVerboseAddressesBalancesData(string addresses)
+        {
+            try
+            {
+                string[] addressesArray = addresses.Split(',');
+
+                this.logger.LogDebug("Asking data for {0} addresses.", addressesArray.Length);
+
+                VerboseAddressBalancesResult result = this.addressIndexer.GetVerboseAddressBalancesData(addressesArray);
+
```

Comment:
Add a log after the result has been returned?

`this.logger.LogDebug("Sending data for {0} addresses.", result.Balances.Count);` or something similar.

Reply:
It will be automatically generated by fody as an exit log